Added route and controller for getting porduct by id

BenFaruna opened the pull request:
you can check this one that was used for the previous cohort

# Fixes Issue/Linear Ticket

> This could be an existing issue or a linear ticket
>
> - Github Issue Example: My PR Closes #{ISSUE}
> - Linear Ticket Example: Fixes ID-#{ISSUE}

# Changes proposed

> Talk about the things you did eg. files changes, dependencies installed e.t.c

# Check List (Check all the applicable boxes)

:rotating_light:Please review the [style guide for contributing](add link here) and [guidelines for contributing](add link here) to this repository.

- [x] My code follows the code style of this project.
- [x] This PR does not contain plagiarized content.
- [x] The title and description of the PR is clear and explains the approach.
- [x] I am making a pull request against the **main branch** (left side).
- [x] My commit messages styles matches our requested structure.
- [x] My code additions will fail neither code linting checks nor unit test.
- [x] I am only making changes to files I was requested to.

# Screenshots/ Videos

When the product id cannot be found
![localhost_8080_api_shop_s  - zuriportfolio-commander-backend - Visual Studio Code 10_14_2023 8_24_44 AM](https://github.com/hngx-org/zuriportfolio-commander-backend/assets/57325382/8c08be05-3638-41fa-807c-cf61a7c0ea3a)

When the product id exists
![localhost_8080_api_shop_s  - zuriportfolio-commander-backend - Visual Studio Code 10_14_2023 8_24_37 AM](https://github.com/hngx-org/zuriportfolio-commander-backend/assets/57325382/0b579b21-f60a-4a2a-b331-4630f098bfe3)

I will be doing the swaggger docs next.

<!-- If the changes are static page changes or UI changes add screenshots -->
<!-- If the changes involve implementing a functionality or working with apis, include a video
detailing how to implement the functionality and the request to the api and responses from the api endpoint-->
<!-- Add all the screenshots/videos which support your changes i.e before your change and after your change -->

Benrobo commented on src/controller/product.controller.ts:
```diff
@@ -280,6 +280,45 @@ export default class ProductController extends BaseController {
+        description: product.description,
+      };
+    } else {
+      return this.error(res, '--product/missing-product', 'Product not found.', 404, null)
```
Your else statement isn't needed.. when you query to get a product by an id, that when you check if that product is available or not and send a 404 error before trying any manipulation after getting the product, pls update that.

BenFaruna commented:
I have fixed that @Benrobo, I also added swagger, now I'm seeing a merge conflict on swagger, I fixed that too

![Swagger UI - Google Chrome 10_14_2023 8_48_31 AM](https://github.com/hngx-org/zuriportfolio-commander-backend/assets/57325382/8d3f4b06-4f10-417b-91a4-8fd697cd5e00)

![Swagger UI - Google Chrome 10_14_2023 8_48_37 AM](https://github.com/hngx-org/zuriportfolio-commander-backend/assets/57325382/3b8fafc6-c938-4273-88e5-a701f0f74031)

BenFaruna commented:
Please unmerge @Benrobo
